src/frontend/screens/Settings/Config.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import * as DocumentPicker from 'expo-document-picker';
+import * as FileSystem from 'expo-file-system';
+import {defineMessages, useIntl} from 'react-intl';
+import {StyleSheet, View} from 'react-native';
+import {Text} from '../../sharedComponents/Text';
+import {
+  useImportProjectConfig,
+  useProjectSettings,
+} from '../../hooks/server/projects';
+import {Loading} from '../../sharedComponents/Loading';
+import {NativeNavigationComponent} from '../../sharedTypes/navigation';
+import {COMAPEO_BLUE, MEDIUM_GREY} from '../../lib/styles';
+import {Button} from '../../sharedComponents/Button';
+import {convertFileUriToPosixPath} from '../../lib/file-system';
+import {UIActivityIndicator} from 'react-native-indicators';
+import {ErrorBottomSheet} from '../../sharedComponents/ErrorBottomSheet';
+
+const m = defineMessages({
+  navTitle: {
+    id: 'screens.Settings.Config.navTitle',
+    defaultMessage: 'Configuration',
+  },
+  name: {
+    id: 'screens.Settings.Config.name',
+    defaultMessage: 'Config Name:',
+  },
+  projectName: {
+    id: 'screens.Settings.Config.projectName',
+    defaultMessage: 'Project Name:',
+  },
+  created: {
+    id: 'screens.Settings.Config.created',
+    defaultMessage: 'Created {date} at {time}',
+  },
+  importConfig: {
+    id: 'screens.Settings.Config.importConfig',
+    defaultMessage: 'Import Config',
+  },
+});
+
+export const Config: NativeNavigationComponent<'Config'> = ({navigation}) => {
+  const {formatMessage} = useIntl();
+  const {data, isPending} = useProjectSettings();
+  const [importError, setImportError] = React.useState<Error | null>(null);
+
+  const importProjectConfigMutation = useImportProjectConfig();
+
+  React.useEffect(() => {
+    // Prevent back navigation while project creation mutation is pending
+    const unsubscribe = navigation.addListener('beforeRemove', event => {
+      if (!importProjectConfigMutation.isPending) {
+        return;
+      }
+
+      event.preventDefault();
+    });
+
+    return () => {
+      unsubscribe();
+    };
+  }, [navigation, importProjectConfigMutation.isPending]);
+
+  async function importConfigFile() {
+    let result;
+    try {
+      result = await DocumentPicker.getDocumentAsync({
+        copyToCacheDirectory: true,
+        multiple: false,
+      });
+    } catch (_err) {
+      setImportError(_err as Error);
+      return;
+    }
+
+    if (result.canceled) {
+      return;
+    }
+
+    const asset = result.assets[0];
+
+    // Shouldn't happen based on how the library works
+    if (!asset) return;
+
+    importProjectConfigMutation.mutate(convertFileUriToPosixPath(asset.uri), {
+      onSuccess: async () => {
+        await FileSystem.deleteAsync(asset.uri).catch((err: unknown) => {
```

Comment:
This doesn't trigger any UI update or state refresh other than invalidating the query. Should it refetch the project settings data to update the component's state? Or maybe I am missing something

Reply:
created a custom hook that invalidates the project settings: https://github.com/digidem/comapeo-mobile/pull/608/commits/652be3f35d2bafbed4af742ffad805e7d4e9d961